apps/dc_tools/odc/apps/dc_tools/export_md.py

```diff
@@ -0,0 +1,264 @@
+
+def get_grids(dataset, band_grids=None):
+    """
+    'band_grids' specify bands for each grid as in:
+    {
+        default: [swir1, swir2]
+        ir: [ir1, ir2]
+    }
+    All the bands not included in the 'band_grids' considered 'default'
+    Returns grids as in
+    {
+     grids:
+        default:
+            shape: [7731, 7621]
+            transform: [30.0, 0.0, 306285.0, 0.0, -30.0, -1802085.0, 0, 0, 1]
+        ir:
+            shape: [3865, 3810]
+            transform: [60.0, 0.0, 306285.0, 0.0, -60.0, -1802085.0, 0, 0, 1]
+    }
+    """
+
+    if not band_grids:
+        # Assume all measurements belong to default grid
+        geo = native_geobox(dataset, [list(dataset.measurements)[0]])
+        return {'grids': {
+            'default': {
+                'shape': list(geo.shape),
+                'transform': list(geo.transform)
+            }
+        }}
+    else:
+        grids = dict()
+        for grid_name in band_grids:
+            geo = native_geobox(dataset, [band_grids[grid_name][0]])
+            grids[grid_name] = {
+                'shape': list(geo.shape),
+                'transform': list(geo.transform)
+            }
+        if not band_grids.get('default'):
+            specified_bands = set()
+            for grid in band_grids:
+                specified_bands.update(band_grids[grid])
+            all_bands = set(list(dataset.measurements))
+            default_bands = all_bands - specified_bands
+            if bool(default_bands):
+                geo = native_geobox(dataset, [list(default_bands)[0]])
+                grids['default'] = {
+                    'shape': list(geo.shape),
+                    'transform': list(geo.transform)
+                }
+        return {'grids': grids}
+
+
+def get_measurements(dataset, band_grids=None):
+    """
+    Extract and return measurement paths in a dictionary:
+    Returns
+    {
+      'measurements':
+      {
+        'coastal_aerosol': {
+                             'path': path_name1,
+                             'band': band_number,
+                             'layer': null or 'layer_name'
+                             'grid': 'ir'
+                           },
+        ...
+      }
+    }
+    """
+    grids_map = {m: grid for grid in band_grids for m in band_grids[grid]}
+    measurements = dataset.measurements
+    for m in measurements:
+        if grids_map.get(m):
+            measurements[m]['grid'] = grids_map[m]
+        else:
+            measurements[m]['grid'] = 'default'
+
+    return {'measurements': measurements}
+
+
+def get_properties(dataset, property_offsets=None):
+    """
+    Extract properties and return values in a dictionary:
+    {
+      'properties':
+      {
+        'datetime': time,
+        'odc:creation_datetime': creation_time,
+        ...
+      }
+    }
+    """
+    props = dict()
+    props['datetime'] = [dataset.time.begin, dataset.time.end]
+    props['odc:creation_datetime'] = dataset.indexed_time
```

Comment:
Can we change this to `odc:processing_datetime`

Reply:
Done